be/src/Project/Services/Catalog/Catalog.Api/Services/Categories/CategoriesService.cs

```diff
@@ -256,9 +239,11 @@ public async Task<CategoryServiceModel> CreateAsync(CreateCategoryServiceModel m
                     MediaId = file,
                 };
 
-                _context.CategoryImages.Add(categoryImage.FillCommonProperties());
+                images.Add(categoryImage.FillCommonProperties());
             }
 
+            await _context.CategoryImages.AddRangeAsync(images);
```

Comment:
**suggestion:** let's use the previous implementation because it will add everything in one batch on .SaveContext(). It has one disadvantage that we create additional redundant List<CategoryImage>() and fill it.